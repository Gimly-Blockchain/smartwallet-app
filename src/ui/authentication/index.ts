export {
  AuthenticationConsent,
} from 'src/ui/authentication/container/authenticationConsent'
