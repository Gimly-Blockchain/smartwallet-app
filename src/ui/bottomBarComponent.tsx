import React from 'react'
import { StyleSheet, View, Dimensions, Platform, TouchableOpacity } from 'react-native'
import Svg, {
  Circle,
  Defs,
  G,
  Mask,
  Path,
  Polygon,
  Use,
} from 'react-native-svg'
import { Colors } from '../styles'
import LinearGradient from 'react-native-linear-gradient'
import { BottomTabBarProps } from 'react-navigation'
import { debug } from '../styles/presets'

const styles = StyleSheet.create({
  wrapper: {
    position: 'absolute',
    width: '100%',
    bottom: -50,
  },
  circle: {
    backgroundColor: Colors.joloColor,
    borderRadius: 35,
    position: 'absolute',
  },
  buttonWrapper: {
    position: 'absolute',
    width: '100%',
    top: 0,
    height: 60,
    zIndex: 2,
    flexDirection: 'row',
    ...debug,
  },
})

// ramda?
const parsePath = () => {
  const origPath =
    'M414,38.2468806 L414,109.427911 L414,109.427911 L0,110.544855 L0,38.2468806 C-2.53876144e-15,17.5163202 16.8054396,0.710880602 37.536,0.710880602 L163.503362,0.710880602 L163.503362,0.710880602 C161.782566,5.63732363 160.846285,10.9383896 160.846285,16.4607145 C160.846285,42.5558923 181.753319,63.710216 207.543594,63.710216 C233.333869,63.710216 254.240903,42.5558923 254.240903,16.4607145 C254.240903,10.9383896 253.304622,5.63732363 251.583826,0.710880602 L376.423017,0.710880602 C397.153577,0.710880602 414,17.5163202 414,38.2468806 Z'
  const origWidth = 414
  const { width } = Dimensions.get('window')

  const coordArray = origPath.split(' ')
  coordArray.pop()
  const coordinates = coordArray.map(vals => {
    const coordinates = vals.split(',')
    const paths = coordinates.map(val => {
      let newVal = val
      let pathType
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      if (isNaN(val[0] as any)) {
        pathType = val[0]
        newVal = val.substr(1)
      } else {
        pathType = ''
      }
      const numVal = parseFloat(newVal)
      const scaledVal = numVal * (width / origWidth)
      return `${pathType}${scaledVal}`
    })
    return paths.join(',')
  })
  coordinates.push('Z')
  return coordinates.join(' ')
}

const SVGBar = () => {
  const { width } = Dimensions.get('window')

  return (
    <Svg width={width} height={110} viewBox={`0 0 ${width} 110`}>
      <Defs>
        <Path
          //d="M414,38.2468806 L414,109.427911 L414,109.427911 L0,110.544855 L0,38.2468806 C-2.53876144e-15,17.5163202 16.8054396,0.710880602 37.536,0.710880602 L163.503362,0.710880602 L163.503362,0.710880602 C161.782566,5.63732363 160.846285,10.9383896 160.846285,16.4607145 C160.846285,42.5558923 181.753319,63.710216 207.543594,63.710216 C233.333869,63.710216 254.240903,42.5558923 254.240903,16.4607145 C254.240903,10.9383896 253.304622,5.63732363 251.583826,0.710880602 L376.423017,0.710880602 C397.153577,0.710880602 414,17.5163202 414,38.2468806 Z"
          d={parsePath()}
          id="path-1"
        />
      </Defs>
      <G id="Symbols" fill="none" fillRule="evenodd">
        <G id="balck_bckg">
          <Mask id="mask-2" fill="white">
            <Use xlinkHref="#path-1" />
          </Mask>
          <G id="Clip-86" />
          <Polygon
            id="Fill-85"
            fill="#0B030D"
            mask="url(#mask-2)"
            points="0 -5.77549882 416.208 -5.77549882 416.208 109.272 6.16191739e-14 109.272"
          />
        </G>
      </G>
    </Svg>
  )
}

interface Props extends BottomTabBarProps {}

export const BottomBarComponent = (props: Props) => {
  console.log(props)
  const { width } = Dimensions.get('window')
  const buttonMarginModif = 16 / 414
  const buttonSizeModif = 0.175
  const circleSize = width * buttonSizeModif

  const buttonVerticalPosition = -Math.floor(
    circleSize / 2 - width * buttonMarginModif,
  )
  const buttonHorizPosition = Math.ceil((width - circleSize) / 2)

  return (
    <View style={styles.wrapper}>
      <View style={styles.buttonWrapper}>
        <View
          style={{ flex: 1, marginLeft: '6%', marginRight: '12%', ...debug }}
        />
        <View
          style={{ flex: 1, marginLeft: '12%', marginRight: '6%', ...debug }}
        />
      </View>
      <TouchableOpacity
        activeOpacity={0.4}
        style={{
          ...styles.circle,
          width: circleSize,
          height: circleSize,
          left: buttonHorizPosition,
          top: buttonVerticalPosition,
        }}
      >
        <LinearGradient
          style={{ flex: 1, borderRadius: 35 }}
          start={{ x: 0, y: 0 }}
          end={{ x: 1, y: 0 }}
          colors={['rgb(145, 25, 66)', 'rgb(210, 45, 105)']}
        />
      </TouchableOpacity>
      <SVGBar />
    </View>
  )
}
