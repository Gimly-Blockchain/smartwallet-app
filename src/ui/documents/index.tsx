export { Documents } from 'src/ui/documents/containers/documents'
export { DocumentDetails } from 'src/ui/documents/containers/documentDetails'
