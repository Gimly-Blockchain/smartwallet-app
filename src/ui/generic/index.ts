export { Exception } from './exception'
export { LoadingScreen } from './loading'
export { LoadingSpinner } from './loadingSpinner'
export { SlidingTabBar } from './slidingTabBar'
