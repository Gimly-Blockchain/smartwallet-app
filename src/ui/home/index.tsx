export { Claims } from 'src/ui/home/containers/claims'
export { ClaimDetails } from 'src/ui/home/containers/claimDetails'
export { Records } from 'src/ui/home/containers/records'
export { ShareConsent } from 'src/ui/sso/containers/shareConsent'
export { CredentialReceive } from 'src/ui/home/containers/credentialReceive'
