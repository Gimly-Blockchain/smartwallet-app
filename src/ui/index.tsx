export { RegistrationProgress } from './registration/containers/loading'
