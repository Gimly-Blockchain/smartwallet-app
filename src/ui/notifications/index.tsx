import React from 'react'
import { RootState } from '../../reducers'
import { connect } from 'react-redux'
import { View } from 'react-native'
import { ThunkDispatch } from '../../store'
import { TopBarNotification } from './topBarNotification'
import {
  Notification,
  NotificationScope,
} from '../../reducers/notifications/types'

interface WithNotificationsProps
  extends ReturnType<typeof mapStateToProps>,
    ReturnType<typeof mapDispatchToProps> {}

/**
 * Reference implementation of a Notification Manager HOC. It will subscribe to the redux state to receive notifications,
 * and after filtering them, will render a {@link TopBarNotification} for each of them.
 * @typeparam P - The props of the wrapped component / container
 * @param WrappedComponent - Notification bars will render in the top part of the wrapped component passed here
 */

const WithTopBarNotificationHOC = <P extends object>(
  WrappedComponent: React.ComponentType<P>,
): React.ComponentType<P & WithNotificationsProps> => ({
  notifications,
  handleConfirm,
  handleDismiss,
  ...rest
}: WithNotificationsProps & P) => {
  const notification = notifications[0]
  return (
    <View
      style={{
        height: '100%',
        width: '100%',
      }}
    >
      {notification && (
        <TopBarNotification
          notification={notification}
          handleConfirm={handleConfirm}
          handleDismiss={handleDismiss}
        />
      )}
      <WrappedComponent {...(rest as P)} />
    </View>
  )
}
// Only global notifications are rendered here
const mapStateToProps = (state: RootState) => ({
  notifications: state.notifications.filter(
    ({ scope }) => scope === NotificationScope.global,
  ),
})

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  handleConfirm: (notification: Notification) =>
    dispatch(notification.handleConfirm(notification)),
  handleDismiss: (notification: Notification) =>
    dispatch(notification.handleDismiss(notification)),
})

export const withTopBarNotifications = <T extends object>(
  toWrap: React.ComponentType<T>,
) =>
  connect(
    mapStateToProps,
    mapDispatchToProps,
  )(WithTopBarNotificationHOC(toWrap))
