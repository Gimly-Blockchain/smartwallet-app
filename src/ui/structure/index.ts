export { Container } from './container'
export { CardWrapper } from './cardWrapper'
export { JolocomButton } from './jolocomButton'
