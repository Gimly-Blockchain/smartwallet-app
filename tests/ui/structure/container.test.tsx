import React from 'react'
import { Wrapper } from 'src/ui/structure/'
import { StyleSheet } from 'react-native'

const ShallowRenderer = require('react-test-renderer/shallow')

describe('generic container component', () => {
  it('matches the snapshot with no extra props', () => {
    const renderer = new ShallowRenderer()
    const rendered = renderer.render(<Wrapper> null </Wrapper>)
    expect(rendered).toMatchSnapshot()
  })

  it('matches the snapshot with style prop', () => {
    const renderer = new ShallowRenderer()
    const styles = StyleSheet.create({
      block: {
        width: '30%',
        backgroundColor: 'blue',
      },
    })

    const rendered = renderer.render(
      <Wrapper style={styles.block}>null</Wrapper>,
    )
    expect(rendered).toMatchSnapshot()
  })
})
