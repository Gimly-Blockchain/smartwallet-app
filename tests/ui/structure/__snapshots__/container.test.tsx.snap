// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic container component matches the snapshot with no extra props 1`] = `
<SafeAreaView
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      },
      undefined,
    ]
  }
>
   null 
</SafeAreaView>
`;

exports[`generic container component matches the snapshot with style prop 1`] = `
<SafeAreaView
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      },
      Object {
        "backgroundColor": "blue",
        "width": "30%",
      },
    ]
  }
>
  null
</SafeAreaView>
`;
